src/boot/efi/boot.c

```diff
@@ -2292,6 +2328,131 @@ static void config_entry_add_linux(
         }
 }
 
+static EFI_STATUS secure_boot_enroll(EFI_FILE *root_dir, Config *config, ConfigEntry *entry) {
+        assert(root_dir);
+        assert(config);
+        assert(entry);
+
+        typedef struct {
+                const CHAR16 *name;
+                const CHAR16 *filename;
+                const EFI_GUID vendor;
+                CHAR8 *buffer;
+                UINTN size;
+        } SecureBootVariable;
+
+        EFI_STATUS err = EFI_SUCCESS;
+        _cleanup_freepool_ CHAR16 *keys_dir = NULL;
+        _cleanup_(file_closep) EFI_FILE *dir = NULL;
+        UINT32 sb_vars_opts =
+                EFI_VARIABLE_NON_VOLATILE |
+                EFI_VARIABLE_BOOTSERVICE_ACCESS |
+                EFI_VARIABLE_RUNTIME_ACCESS |
+                EFI_VARIABLE_TIME_BASED_AUTHENTICATED_WRITE_ACCESS;
+        UINT64 key;
+
+        clear_screen(COLOR_NORMAL);
+        ps_string(L"Enrolling secure boot keys from directory: %s\n", entry->current_name);
+
+        Print(L"*** DANGER ***\n");
+        Print(L"Enrolling Secure Boot keys might soft-brick your machine.\n");
+        Print(L"However the option i-might-brick-my-machine-please-auto-enroll-keys has been set\n");
+        Print(L"The enrollement of secure boot keys will proceed after %d seconds\n", config->timeout_sec);
+        Print(L"*** DANGER ***\n");
+
+        Print(L"\nType q, Q or ESC to cancel.\n\n");
```

Comment:
maybe reverse this. expect that people type in "k" or so if they want to proceed.

(why "k"? because it sounds like "ok", and is on the same key regardless if you use qwertz, qwerty or azerty keymaps)